### ISSUE TITLE
Open links in current desktop space

### ISSUE THREAD

**Issue by andreiborisov** [deanishe/alfred-safari-assistant#23]
I've organised my workspace as such that I have different desktop spaces for different activities/projects. Naturally, I have Safari windows opened in different spaces.

The problem is, when I try to use Safari Assistant in this context, it opens links in the existing Safari window, even if that window in a different space. I understand, that's more a quirk of macOS itself, but nonetheless it's inconvenient.

So I've made simple URL handler helper app that runs in the background, opens links in the new tab if there is Safari window already present or creates new window if it's not:

```
on open location theURL
	tell application "System Events"
		try
			if (count of windows of process "Safari") = 0 then
				tell application "Safari" to make new document with properties {URL:theURL}
			else
				tell application "Safari" to open location theURL
			end if
		on error
			tell application "Safari" to set URL of front document to theURL
		end try
		tell application "Safari" to activate
	end tell
end open location
```

It works just fine with Alfred itself and most workflows, but since Safari Assistant appears not to use URL scheme to open Safari, my workaround doesn't work with it.

What would be the proper way to fix this? ~~Do you think it would be helpful for other users to have such option in Safari Assistant?~~

**Comment by andreiborisov** [comment]
It seems my problem is related to https://github.com/deanishe/alfred-safari-assistant/issues/22. Perhaps, an option to honor system URL handler can be added?

Or there is a possibility to override default action with my own action script?

Any advice would be appreciated🥺

**Comment by deanishe** [comment]
> It seems my problem is related to #22

Yeah. If you'd commented on that issue, I probably wouldn't have forced Safari.

It'd be relatively easy to add an option to specify an application to open URLs in, and your reason for wanting the option makes a lot of sense.

It's unlikely that I'll ever make the workflow Space-aware, however. Especially as the workflow makes it pretty simple to add an "Open URL in New Window" action. You could add your script fairly easily to the workflow.

**Comment by andreiborisov** [comment]
> Yeah. If you'd commented on that issue, I probably wouldn't have forced Safari.

I've figured out that it's even possible to fix this macOS quirk just recently and #22 didn't seem like a big change at the time...

> Workflow makes it pretty simple to add an "Open URL in New Window" action. You could add your script fairly easily to the workflow.

I believe that would be the most logical and flexible solution.
